packages/daemon/src/host.js

```diff
@@ -380,11 +395,22 @@ export const makeHostMaker = ({
           )}`,
         );
       }
-      return /** @type {Promise<import('./types.js').EndoHost>} */ (
-        // Behold, recursion:
-        // eslint-disable-next-line no-use-before-define
-        provideValueForFormulaIdentifier(formulaIdentifier)
-      );
+      const newHostController =
+        /** @type {import('./types.js').Controller<>} */ (
+          provideControllerForFormulaIdentifier(formulaIdentifier)
+        );
+      if (introducedNames !== undefined) {
+        const { petStore: newHostPetStore } = await newHostController.internal;
+        for (const [parentName, childName] of Object.entries(introducedNames)) {
+          const introducedFormulaIdentifier =
+            lookupFormulaIdentifierForName(parentName);
+          if (introducedFormulaIdentifier === undefined) {
+            continue;
+          }
+          await newHostPetStore.write(childName, introducedFormulaIdentifier);
+        }
```

Comment:
Opportunity for `Promise.all` and `map`. Not to imply it makes much difference since writes will be serialized.